src/vs/workbench/contrib/chat/browser/chatTerminalCommandMirror.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable, ImmortalReference } from '../../../../base/common/lifecycle.js';
+import type { ITerminalCommand } from '../../../../platform/terminal/common/capabilities/capabilities.js';
+import { ITerminalInstance, ITerminalService, type IDetachedTerminalInstance } from '../../terminal/browser/terminal.js';
+import { DetachedProcessInfo } from '../../terminal/browser/detachedTerminal.js';
+import { IInstantiationService } from '../../../../platform/instantiation/common/instantiation.js';
+import { TerminalInstanceColorProvider } from '../../terminal/browser/terminalInstance.js';
+import { TerminalLocation } from '../../../../platform/terminal/common/terminal.js';
+
+export class DetachedTerminalCommandMirror extends Disposable {
```

Comment:
This can live in `terminal/browser/` instead? Don't want to clutter chat with this, plus there's a good chance we'll reuse it in something else later.